partner_changeset/models/changeset_field_rule.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Authors: Guewen Baconnier
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp import models, fields, api
+from openerp.tools.cache import ormcache
+
+
+class ChangesetFieldRule(models.Model):
+    _name = 'changeset.field.rule'
+    _description = 'Changeset Field Rules'
+    _rec_name = 'field_id'
+
+    field_id = fields.Many2one(
+        comodel_name='ir.model.fields',
+        string='Field',
+        domain="[('model_id.model', '=', 'res.partner'), "
+               " ('ttype', 'in', ('char', 'selection', 'date', 'datetime', "
+               "                 'float', 'integer', 'text', 'boolean', "
+               "                 'many2one')), "
+               " ('readonly', '=', False)]",
+        ondelete='cascade',
+        required=True,
+    )
+    action = fields.Selection(
+        selection='_selection_action',
+        string='Action',
+        required=True,
+        help="Auto: always apply a change.\n"
+             "Validate: manually applied by an administrator.\n"
+             "Never: change never applied.",
```

Comment:
It is different from `readonly` because it will log it in the changeset.
You will never use it in global rules (or user rules I guess), but when you configure rules for other sources, it could come in handy to be able to prevent a source to write in a given field.